### PR TITLE
Importer for casa

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/models/health/epic_base.rb [hunk 1]
```diff
@@ -33,8 +27,9 @@ def clean_row(row:, data_source_id:) # rubocop:disable Lint/UnusedMethodArgument
       row
     end
 
-    def self.use_tsql_import?
-      true
+    def self.process_new_data(values)
+      where(data_source_id: values.first[:data_source_id]).delete_all # Rather than merge, discard old data
```
Comment (eanders) [inline app/models/health/epic_base.rb:33]: might want a `return unless values.present?`

File: app/models/health/tasks/import_epic.rb [hunk 2]
```diff
@@ -26,29 +25,33 @@ def initialize(logger: Rails.logger, load_locally: false, configs: nil, prevent_
...
       @new_patients = []
-      @configs = configs
+      @configs = configs || Health::ImportConfig.all
       @prevent_massive_change = prevent_massive_change
+
+      raise 'load_locally only supports a single config' if @load_locally && @configs.keys.count > 1
     end
 
     def run!
-      @configs ||= YAML::load(ERB.new(File.read(Rails.root.join("config","health_sftp.yml"))).result)[Rails.env]
```
Comment (eanders) [inline app/models/health/tasks/import_epic.rb:34]: Nice! does this mean we can remove this from .env?

Comment (awisema) [reply app/models/health/tasks/import_epic.rb:34]: Yes... after the migration is run to move the contents into the DB :-)

File: app/models/health/tasks/import_epic.rb [hunk 3]
```diff
@@ -163,9 +162,9 @@ def fetch_files
 
     def read_csv_file path:
       # Look at the file to see if we can determine the encoding
-      @file_encoding = CharlockHolmes::EncodingDetector
-        .detect(File.read(path))
-        .try(:[], :encoding)
+      @file_encoding = CharlockHolmes::EncodingDetector.
+        detect(File.read(path)).
+        try(:[], :encoding)
```
Comment (eanders) [inline app/models/health/tasks/import_epic.rb:167]: Maybe not worth it, but I wonder if we should use the CSV reader we built for the importer.

Comment (awisema) [reply app/models/health/tasks/import_epic.rb:167]: If we generalize this, I'd say yes -- but unless the files from Casa turns out to be problematic, I'm inclined to leave this as it is...

File: app/models/health/vaccination.rb [hunk 4]
```diff
@@ -190,10 +190,6 @@ def self.propagate_to_clients
       GrdaWarehouse::HealthEmergency::Vaccination.import(new_vaccinations) if new_vaccinations.present?
     end
 
-    def self.use_tsql_import?
```
Comment (eanders) [inline app/models/health/vaccination.rb:193]: One of these was true right?  I feel like I recently had tried extricating it, which lead to this, my memory is shaky on why.

Comment (awisema) [reply app/models/health/vaccination.rb:193]: Yes -- `Health::EpicBase` had it set to true.

**Pull request merged**